### PR TITLE
Update to TileDB 2.1

### PR DESCRIPTION
**Pull request by aaronwolen**

### CODE REVIEW

File: libtiledbvcf/cmake/Modules/FindTileDB_EP.cmake [hunk 1]
```diff
@@ -47,8 +47,8 @@ else()
 
     ExternalProject_Add(ep_tiledb
       PREFIX "externals"
-      URL "https://github.com/TileDB-Inc/TileDB/archive/2.0.8.zip"
-      URL_HASH SHA1=f0ba8f768aafe844530f85f3c8b51ac2d27d7194
+      URL "https://github.com/TileDB-Inc/TileDB/archive/2.1.0-rc1.tar.gz"
+      URL_HASH SHA1=1ffbad9f88e18f55a111d030aa6d3ce49bcf607f5a14d246ffc99fd5aaa820f8
```
Comment (Shelnutt2) [inline libtiledbvcf/cmake/Modules/FindTileDB_EP.cmake:51]: This looks like a SHA256 not a SHA1

Comment (aaronwolen) [reply libtiledbvcf/cmake/Modules/FindTileDB_EP.cmake:51]: Ha, right. I'm still in conda mode.

**Comment by Shelnutt2** [comment]
@aaronwolen can you rebase on master to get the CLI test diff output and also update to 2.1.0-rc2?

File: apis/python/tests/test_tiledbvcf.py [hunk 2]
```diff
@@ -412,8 +412,8 @@ def test_read_config():
 
     cfg = tiledbvcf.ReadConfig(memory_budget_mb=512,
                                region_partition=(0, 3),
-                               tiledb_config=['sm.tile_cache_size=0',
-                                              'sm.num_reader_threads=1'])
+                               tiledb_config=['sm.tile_cache_size=1',
```
Comment (Shelnutt2) [inline apis/python/tests/test_tiledbvcf.py:415]: Should we change this back to 0?

**Review by Shelnutt2 — CHANGES_REQUESTED** [review]

File: apis/python/tests/test_tiledbvcf.py (hunk 2 above)
Comment (aaronwolen) [reply apis/python/tests/test_tiledbvcf.py]: Done.

**Review by Shelnutt2 — APPROVED** [review]

**Pull request merged**